### ISSUE TITLE
Potential error when calculating dihedrals with top.select

### ISSUE THREAD

**Issue by rcrehuet** [mdtraj/mdtraj#1330]
Because the `select` method from `top` does not conserve the atom order in the argument string, the dihedral angle obtained can easily be wrong. I mean, when one selects with:

    t.top.select('resid 10 and name OE1 CD NE2 HE21')

one obtains the same array of atoms (with the same order) than when you select:

        t.top.select('resid 10 and name CD OE1 NE2 HE21')

Thus, the following dihedrals should be different (by 180º) and they are the same.
    
    md.compute_dihedrals(t, [t.top.select('resid 10 and name OE1 CD NE2 HE22')])
    md.compute_dihedrals(t, [t.top.select('resid 10 and name CD OE1 NE2 HE22')])

What would be the recommended workaround for this? Would it be possible to chage `select` so that it returns the atoms in the order specified in its argument string?

**Comment by chrisbarnettster** [comment]
A workaround, a little hacky :bowtie: :

```
def ordered_select(topology,firstselect, orderedatoms=["O5","C5","C6","O6"]):
    if len(set(orderedatoms))<len(orderedatoms):
        print("Duplicate atoms")
        return[]
    try:
        return [topology.select(" ".join([firstselect,i]))[0] for i in orderedatoms]
    except Exception as e:
        print(e)
        return []
```
Atom order is conserved and the example above will now work as expected:

```
md.compute_dihedrals(t, [ordered_select(t.top,'resid 10 and name',['OE1','CD','NE2', 'HE22'])])
md.compute_dihedrals(t, [ordered_select(t.top,'resid 10 and name',['CD','OE1','NE2', 'HE22'])])
```

TLDR;

- invalid selections will return empty [] as per what happens with the usual select. For invalid selections, the error "index 0 is out of bounds for axis 0 with size 0" is printed.
- When an empty selection is passed on to compute_dihedrals, this will result in "indices must be shape (Any, 4). You supplied  (1, 0)" or similar. 
- In the context no duplicate atoms should exist in the selection, thus the short check for duplicates at the beginning.

**Comment by chrisbarnettster** [comment]
This can be done without a helper function, although duplicate atoms are then possible. See information on ordered selections - https://www.mdanalysis.org/docs/documentation_pages/selections.html#ordered-selections. 


> If the ordering of atoms is crucial (for instance when describing angles or dihedrals) or if duplicate atoms are required then one has to concatenate multiple AtomGroups, which does not sort them.
>The most straightforward way to concatenate two AtomGroups is by using the + operator:
> 
> 
> ```
> >>> ordered = u.select_atoms("segid DMPC and resid 3 and name P") + u.select_atoms("segid DMPC and resid 2 and name P")
> >>> print list(ordered)
> ```
> [< Atom 570: name 'P' of type '180' of resid 'DMPC', 3 and 'DMPC'>,
> < Atom 452: name 'P' of type '180' of resid 'DMPC', 2 and 'DMPC'>]
> 
> 
> A shortcut is to provide two or more selections to select_atoms(), which then does the concatenation automatically:
> 
> ```
> >>> print list(universe.select_atoms("segid DMPC and resid 3 and name P", "segid DMPC and resid 2 and name P"))
> ```
> [< Atom 570: name 'P' of type '180' of resid 'DMPC', 3 and 'DMPC'>,
> < Atom 452: name 'P' of type '180' of resid 'DMPC', 2 and 'DMPC'>]
> 
>
